homeassistant/components/rova/sensor.py

```diff
@@ -96,33 +111,33 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     data_service = RovaData(api)
 
     # Create a new sensor for each garbage type.
-    entities = []
-    for sensor_key in config[CONF_MONITORED_CONDITIONS]:
-        sensor = RovaSensor(platform_name, sensor_key, data_service)
-        entities.append(sensor)
-
+    entities = [
+        RovaSensor(platform_name, description, data_service)
+        for description in SENSOR_TYPES
+        if description.key in config[CONF_MONITORED_CONDITIONS]
+    ]
     add_entities(entities, True)
 
 
 class RovaSensor(SensorEntity):
     """Representation of a Rova sensor."""
 
-    def __init__(self, platform_name, sensor_key, data_service):
+    entity_description: RovaSensorEntityDescription
+
+    def __init__(
+        self, platform_name, description: RovaSensorEntityDescription, data_service
+    ):
         """Initialize the sensor."""
-        self.sensor_key = sensor_key
+        self.entity_description = description
         self.platform_name = platform_name
         self.data_service = data_service
 
         self._state = None
 
-        metadata = SENSOR_TYPES[sensor_key]
-        self._json_key = metadata.json_key
-        self._attr_icon = metadata.icon
-
     @property
     def name(self):
         """Return the name."""
-        return f"{self.platform_name}_{self.sensor_key}"
+        return f"{self.platform_name}_{self.entity_description.key}"
```

Comment:
It's weird that we don't use the description name as suffix instead.

Reply:
True, however changing it would likely be a breaking change.

Reply:
Not if we change the name key to what's used here. It doesn't seem used elsewhere.